### PR TITLE
Update to deal with use of Spring BOM in samples

### PR DESCRIPTION
**Pull request by JimSuplizio**
Note: Before this gets checked in https://github.com/Azure/azure-sdk-for-java/pull/12966 needs to be checked merged into master.

Spring samples require the use of the spring bom and the problem with this is, it's a BOM file and the spring dependencies are pulled in through that which means any dependencies may or may not have versions. Unfortunately, there are still version tags azure sdk client libraries which means these files have to be "sort of" scanned.

Basically I'm doing this
* Look for pom files with a specific parent, all of these parent from azure-spring-boot-sample.
* Inspect all of the dependencies 
  * if any dependency is an azure sdk library then it must have a version 
  * If the dependency has a version element then it must have a version tag
  * If the dependency doesn't have a version element then ignore it

### CODE REVIEW

File: eng/versioning/pom_file_version_scanner.ps1 [hunk 1]
```diff
@@ -248,6 +255,100 @@ function Confirm-Node-Is-Part-Of-Configuration {
...
+                    }
+                } else {
+                    $script:FoundError = $true
+                    Write-Error-With-Color "Error: dependency is missing version tag groupId=$($groupId), artifactId=$($artifactId) tag should be <!-- {x-version-update;$($groupId):$($artifactId);current|dependency|external_dependency<select one>} -->"
+                }
+            }
+        } else {
+            # else, if there's a version tag verify it, otherwise just skip it since the version should be coming
+            # from the bom
+            if ($versionNode -and $versionNode.NextSibling -and $versionNode.NextSibling.NodeType -eq "Comment")
```
Comment (weshaggard) [inline eng/versioning/pom_file_version_scanner.ps1:339]: What is the conditions about the NextSibling? I'd assume if there is a version tag there must also be a comment tag.

Comment (JimSuplizio) [reply eng/versioning/pom_file_version_scanner.ps1:339]: Good catch. This should be checking if there's a version node separately...something like. I'll update it in the next commit.
`if ($versionNode)
    if ($versionNode.NextSibling -and $versionNode.NextSibling.NodeType -eq "Comment") 
    {
        ...do verification on the tag and version
    }
    else {
        Error because there's a version with no version tag
    }
`

Comment (JimSuplizio) [reply eng/versioning/pom_file_version_scanner.ps1]: This was added with the latest commit. I'm resolving the conversation.
